pkg/apis/work/v1alpha2/binding_types.go

```diff
@@ -362,6 +366,41 @@ type AggregatedStatusItem struct {
 	Health ResourceHealth `json:"health,omitempty"`
 }
 
+// FailoverHistoryItem represents either a failover event in the history.
+type FailoverHistoryItem struct {
+	// Reason denotes the type of failover.
+	// +required
+	Reason FailoverReason `json:"reason"`
+
+	// StartTime is the timestamp of when the failover occurred.
+	// +required
+	StartTime metav1.Time `json:"failoverTime"`
+
+	// ClustersBeforeFailover records the cluster where the application was running prior to failover.
+	// +required
+	ClusterBeforeFailover string `json:"originalCluster"`
+
+	// ClustersAfterFailover records the cluster where the application is running after failover.
+	// +optional
+	ClusterAfterFailover string `json:"targetCluster,omitempty"`
```

Comment:
```suggestion
	// ClustersBeforeFailover records the cluster where the application was running prior to failover.
	// +required
	ClusterBeforeFailover []string `json:"clusterBeforeFailover"`

	// ClustersAfterFailover records the cluster where the application is running after failover.
	// +required
	ClusterAfterFailover []string `json:"clusterAfterFailover"`
```
That's my thought.

Reply:
I've updated the type to be `[]TargetCluster` so that it's aligned with the types used in ResourceBinding, just haven't pushed the changes yet. Let me know if that is okay as well -

Reply:
We usually use the field name as the json tag.  I mean `originalCluster` and `targetCluster`.

In addition, do you think `ClusterBeforeFailover` and `ClusterAfterFailover` should be `required` filed?
I'm not sure yet, but I think we can go with `required` for now.

Reply:
Oh, yes this is updated as well! Will reflect once I push my recent changes. :)

Reply:
Thanks, i see.